src/main/java/org/broadinstitute/consent/http/db/mapper/StudyReducer.java

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.consent.http.db.mapper;
+
+import org.broadinstitute.consent.http.enumeration.PropertyType;
+import org.broadinstitute.consent.http.models.FileStorageObject;
+import org.broadinstitute.consent.http.models.Study;
+import org.broadinstitute.consent.http.models.StudyProperty;
+import org.jdbi.v3.core.result.LinkedHashMapRowReducer;
+import org.jdbi.v3.core.result.RowView;
+
+import java.util.Map;
+import java.util.Objects;
+
+public class StudyReducer implements LinkedHashMapRowReducer<Integer, Study>, RowMapperHelper {
+
+  @Override
+  public void accumulate(Map<Integer, Study> map, RowView rowView) {
+    Study study =
+            map.computeIfAbsent(
+                    rowView.getColumn("study_id", Integer.class), id -> rowView.getRow(Study.class));
+
+    reduceStudy(study, rowView);
+  }
+
+
+  public void reduceStudy(Study study, RowView rowView) {
```

Comment:
Note: doing this allowed me to use the reducer in a more generic way - the reducer could be used both when reducing the study field for a dataset in the DatasetDAO and for reducing normal study objects.